xmipptomo/protocols/protocol_score_transform.py

```diff
@@ -0,0 +1,138 @@
+        self.dist = [(t1[0], quaternion_distance(t1[1], t2[1]))
+                     for t1, t2 in zip(first_quaternions, second_quaternions)
+                     if t1[0] == t2[0]]
+
+        # Save summary to use it in the protocol Info
+        only_distances = np.asarray(list(zip(*self.dist))[1])
+        mean_dist = np.mean(only_distances)
+        std_dist = np.std(only_distances)
+        percentage_outliers = np.sum(only_distances > mean_dist + 3 * std_dist) \
+                              + np.sum(only_distances < mean_dist - 3 * std_dist)
+        percentage_outliers = 100 * percentage_outliers / len(only_distances)
+        summary = self._getExtraPath('Summary.txt')
+        with open(summary, 'w') as file:
+            file.write('Mean distance between the two sets: %.2f\n' % mean_dist)
+            file.write('Estimated percentage of outliers: %.2f%%\n' % percentage_outliers)
+
+    def createOutputStep(self):
+        outSubtomos = self._createSetOfSubTomograms()
+        outSubtomos.setSamplingRate(self.second_subtomos.getSamplingRate())
+        outSubtomos.setCoordinates3D(self.second_subtomos.getCoordinates3D())
+        # acquisition = TomoAcquisition()
+        # acquisition.setAngleMin(self.second_subtomos.getFirstItem().getAcquisition().getAngleMin())
+        # acquisition.setAngleMax(self.second_subtomos.getFirstItem().getAcquisition().getAngleMax())
+        # acquisition.setStep(self.second_subtomos.getFirstItem().getAcquisition().getStep())
+        # outSubtomos.setAcquisition(acquisition)
+        for ids, inSubtomo in enumerate(self.first_subtomos.iterItems()):
+            subtomogram = SubTomogram()
+            subtomogram.setObjId(self.dist[ids][0])
+            subtomogram.setLocation(inSubtomo.getLocation())
+            subtomogram.setCoordinate3D(inSubtomo.getCoordinate3D())
+            subtomogram.setTransform(inSubtomo.getTransform())
+            subtomogram.setVolName(inSubtomo.getVolName())
+            subtomogram.distanceScore = Float(self.dist[ids][1])
+            outSubtomos.append(subtomogram)
+        self._defineOutputs(outputSetOfSubtomogram=outSubtomos)
+        self._defineSourceRelation(self.firstSubtomos, outSubtomos)
+        self._defineSourceRelation(self.secondSubtomos, outSubtomos)
+
+    # --------------------------- UTILS functions ---------------------------
+    def queryMatrices(self, subtomos):
+        matrices = [(subtomo.getObjId(), subtomo.getTransform().getMatrix())
+                    for subtomo in subtomos.iterItems()]
+        return matrices
+
+    # --------------------------- INFO functions ---------------------------
+    def _summary(self):
+        summary = []
+
+        if self.getOutputsSize() >= 1:
+            summary_file = self._getExtraPath('Summary.txt')
+            with open(summary_file, 'r') as file:
+                summary.append(file.read())
+        else:
+            summary.append('Output not ready yet')
+
+        return summary
+
+
+
```

Comment:
I think it is a good idea to add a validation function to check that both input SetOfSubtomograms have transformation matrix. Something like: 
`def _validate(self):`
 `       validateMsgs = []`
     `   for subtomo in self.input.get().iterItems():`
        `    if not subtomo.hasTransform():`
             `   validateMsgs.append('Please provide subtomograms which have transformation matrix in "inputAlignment".')`
      `  return validateMsgs`